internal/fs/buffer/inmemory_write_buffer_test.go

```diff
@@ -41,24 +46,240 @@ func (t *MemoryBufferTest) SetUp(ti *TestInfo) {
 
 func (t *MemoryBufferTest) TearDown() {}
 
+// //////////////////////////////////////////////////////////////////////
+// Helpers
+// //////////////////////////////////////////////////////////////////////
+
+func generateRandomData(size int64) []byte {
+	r := rand.New(rand.NewSource(time.Now().UnixNano()))
+	data := make([]byte, size)
+	r.Read(data)
+	return data
+}
+
 // //////////////////////////////////////////////////////////////////////
 // Tests
 // //////////////////////////////////////////////////////////////////////
 
 func (t *MemoryBufferTest) TestCreateEmptyInMemoryBuffer() {
 	t.mb = CreateInMemoryWriteBuffer()
 
-	AssertEq(nil, t.mb.buffer)
+	AssertEq(nil, t.mb.currentBuffer)
+	AssertEq(nil, t.mb.flushedBuffer)
 	AssertEq(0, ChunkSize)
 }
 
 func (t *MemoryBufferTest) TestInitializeInMemoryBuffer() {
-	bufferSizeMB := 10
+	sizeInMB := 1
 	t.mb = CreateInMemoryWriteBuffer()
 
-	t.mb.InitializeBuffer(bufferSizeMB)
+	t.mb.InitializeBuffer(sizeInMB)
+
+	AssertEq(sizeInMB*MiB, ChunkSize)
+	AssertEq(sizeInMB*MiB, cap(t.mb.currentBuffer))
+	AssertEq(0, len(t.mb.currentBuffer))
+	AssertEq(0, t.mb.currentBufferStartOffset)
+	AssertEq(sizeInMB*MiB, t.mb.currentBufferEndOffset)
```

Comment:
As we have already asserted on `ChunkSize == sizeInMB*MiB`

```suggestion
	AssertEq(ChunkSize, t.mb.currentBufferEndOffset)
```